### ISSUE TITLE
GHCJS 8.4: ReferenceError: h$store_load_barrier is not defined

### ISSUE THREAD

**Issue by Xitian9** [ghcjs/ghcjs#705]
I'm using Reflex together with Streamly to perform some asynchronous I/O. In GHCJS-8.4.0.1, this results in the above error.

Streamly depends on atomic-primops which defines storeLoadBarrier, so I suspect the issue goes back to that.

Here is a minmial working example
```
module MWE where

import Reflex.Dom

import           Control.Monad.IO.Class (liftIO)
import qualified Streamly               as S
import qualified Streamly.Prelude       as S


runMain :: IO ()
runMain = mainWidget bodyElement

bodyElement :: MonadWidget t m => m ()
bodyElement = do
    eStart <- getPostBuild
    eGame <- consumeStream eStart
    dGame <- holdDyn False eGame
    display dGame
    return ()

-- | Consume items from a stream whenever an event is fired
consumeStream :: MonadWidget t m => Event t () -> m (Event t Bool)
consumeStream ev = do
    let xs = S.uncons . S.asyncly . S.repeatM $ return True :: IO (Maybe (Bool, S.Async Bool))
    eList <- performEvent $ liftIO xs <$ ev
    return $ fmapMaybe (fmap fst) eList
```

**Comment by harendra-kumar** [comment]
Hi @Xitian9, just saw this issue today. I think the barriers are not needed when using it in javascript because javascript uses only a single processor. We can support a ghcjs build for streamly compiling out these SMP primitives. Of course, you can still use multiple-threads and async IO.

**Comment by Xitian9** [comment]
In that case I'll close this issue as not within the scope of ghcjs.

**Comment by cdsmith** [comment]
If I am understanding correctly, I think it's a mistake to close this issue.  If GHCJS doesn't need load barriers (and I agree it doesn't), this it should implement them as no-ops.  The goal has always been for GHCJS to build as much of hackage as is reasonable without special adaptations.

**Comment by harendra-kumar** [comment]
I agree, the low level primitives can be implemented as noop on the JS platform so that the higher level libraries remain portable across all platforms.
